sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/EmbeddingItem.java

```diff
@@ -64,4 +65,32 @@ public List<Double> getEmbedding() {
     public int getPromptIndex() {
         return this.promptIndex;
     }
+
+    /**
+     * Creates an instance of EmbeddingItem class.
+     *
+     * @param embedding the embedding value to set.
+     * @param promptIndex the promptIndex value to set.
+     */
+    @JsonCreator
+    private EmbeddingItem(@JsonProperty(value = "embedding") BinaryData embedding,
+        @JsonProperty(value = "index") int promptIndex) {
+        this.embedding = embedding;
+        this.promptIndex = promptIndex;
+        embeddingFloatList = convertBase64ToFloatList(embedding.toString());
```

Comment:
Doing this conversion in the constructor will be eagerly converting the base64 string to float array. Can we do this lazily only when `getEmbedding` method is called?

Reply:
done